File: sources/julia/12_matrix_add.jl
# task 12 matrix_add — expected output: 999000000
# build: julia 12_matrix_add.jl    run: julia 12_matrix_add.jl

function main()
    n = 1000
    a = Vector{Int64}(undef, n * n)
    b = Vector{Int64}(undef, n * n)
    c = Vector{Int64}(undef, n * n)

    for i in 0:(n - 1)
        for j in 0:(n - 1)
            idx = i * n + j + 1
            a[idx] = i + j
            b[idx] = i - j
        end
    end

    for idx in 1:(n * n)
        c[idx] = a[idx] + b[idx]
    end

    total = Int64(0)
    for idx in 1:(n * n)
        total += c[idx]
    end
    println(total)
end

main()

File: sources/go/12_matrix_add.go
// task 12 matrix_add — expected output: 999000000
// build: go build -o prog 12_matrix_add.go    run: ./prog
// note: plain "go build" only; do not set GOGC=off or any other tuning flags
package main

import "fmt"

func main() {
	const n = 1000
	a := make([]int64, n*n)
	b := make([]int64, n*n)
	c := make([]int64, n*n)

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			a[i*n+j] = int64(i + j)
			b[i*n+j] = int64(i - j)
		}
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			c[i*n+j] = a[i*n+j] + b[i*n+j]
		}
	}

	var total int64
	for _, v := range c {
		total += v
	}
	fmt.Println(total)
}
